### PR TITLE
GitHub issue template for documentation internal to `dbt-core`

### LINKED ISSUE #10385: issue template for docs.getdbt.com should disambiguate internal docs like CONTRIBUTING and ARCHITECHTURE
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

#10384 exposed a shortcoming of our current New Issue guidance. There's an entry for "Documentation" with the descritption:

> Problems and issues with dbt documentation

Clicking it redirects you to create a new issue on the docs.getdbt.com repo.

However, in this instance, @farmanp actually wanted to modify a dbt-core internal document: `ARCHITECTURE.md`.

This is how things are currently configured

https://github.com/dbt-labs/dbt-core/blob/c215697a02d7c452c64d4d66f7ba77cf6f2181b6/.github/ISSUE_TEMPLATE/config.yml#L2-L5

image

### Expected Behavior

~~Perhaps we modify the link description to say something like~~

> ~~Problems and issues with dbt documentation hosted on `docs.getdbt.com`. Issues with `.md` that live in this repo (`README`, `CONTRIBUTING`, `ARCHITECTURE`) should be opened with `bugs` label~~

Actually I propose a new template option: "code docs" (working title). Issues created with this template would have a specific label applied and the description would say something like:

> Issues with Markdown files that live in this repo (`README`, `CONTRIBUTING`, `ARCHITECTURE`)

#10386 would be a great candidate for this new designation

### Steps To Reproduce

see above

### Relevant log output

_No response_

### Environment

```markdown
- OS:
- Python:
- dbt:
```

### Which database adapter are you using with dbt?

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by mirnawong1**
clarify documentation issue relates to dbt product docs.

resolves #10385

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -24,3 +24,4 @@ contact_links:
   - name: Create an issue for dbt-spark
     url: https://github.com/dbt-labs/dbt-spark/issues/new/choose
     about: Report a bug or request a feature for dbt-spark
+
```
Comment (dbeatty10) [inline .github/ISSUE_TEMPLATE/config.yml:27]: Removing this line to satisfy the "Fix End of Files" CI check

```suggestion
```

**Review by dbeatty10 — APPROVED** [review]
Looks good to delineate issues related to product documentation vs. docs internal to this repo.

**Review by aranke — APPROVED** [review]
LGTM, ty!

**Comment by mirnawong1** [comment]
thanks so much team!

**Pull request merged**